lib/rucio/core/rule.py

```diff
@@ -362,19 +386,40 @@ def add_rule(dids, account, copies, rse_expression, grouping, weight, lifetime,
                 new_rule.state = RuleState.STUCK
                 new_rule.error = 'MissingSourceReplica'
                 if new_rule.grouping != RuleGrouping.NONE:
-                    session.query(models.DatasetLock).filter_by(rule_id=new_rule.id).update({'state': LockState.STUCK})
+                    stmt = update(
+                        models.DatasetLock
+                    ).where(
+                        models.DatasetLock.scope == new_rule.scope
+                    ).values(
+                        {'state': LockState.STUCK}
+                    )
+                    session.execute(stmt)
```

Comment:
The filter condition is different here:
- old: `rule_id=new_rule.id`
- new: `models.DatasetLock.scope == new_rule.scope`

The result is probably the same, but it would be great if you could clarify why this was changed

Reply:
You're right, changed this to use the same filter conditions as previously:

`models.DatasetLock.rule_id == new_rule.id`